crates/hyperswitch_domain_models/src/router_data.rs

```diff
@@ -1070,3 +1073,213 @@ impl
         }
     }
 }
+
+#[cfg(feature = "v2")]
+impl
+    TrackerPostUpdateObjects<
+        router_flow_types::SetupMandate,
+        router_request_types::SetupMandateRequestData,
+        payments::PaymentConfirmData<router_flow_types::SetupMandate>,
+    >
+    for RouterData<
+        router_flow_types::SetupMandate,
+        router_request_types::SetupMandateRequestData,
+        router_response_types::PaymentsResponseData,
+    >
```

Comment:
This is almost entirely a copy-paste of the implementation of the Autorize flow. Didn't bother to reduce code duplication between these two implementations.